gateway/src/main.rs

```diff
@@ -58,7 +58,7 @@ async fn main() -> io::Result<()> {
 }
 
 async fn start(db: SqlitePool, fs: PathBuf, args: StartArgs) -> io::Result<()> {
```

Comment:
Do we still need this to be in a separate function? But maybe also in another restructure PR.

Reply:
Yes I think so, this one is growing quite large. I will make a note of it.